streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,216 @@
+
+        final ChannelBuilder channelBuilder = ClientUtils.createChannelBuilder(streamsConfig.values());
+
+        final Selector selector = new Selector(streamsConfig.getLong(StreamsConfig.CONNECTIONS_MAX_IDLE_MS_CONFIG), metrics, time, "kafka-client", channelBuilder);
+
+        kafkaClient = new NetworkClient(
+                selector,
+                metadata,
+                streamsConfig.getString(StreamsConfig.CLIENT_ID_CONFIG),
+                MAX_INFLIGHT_REQUESTS, // a fixed large enough value will suffice
+                streamsConfig.getLong(StreamsConfig.RECONNECT_BACKOFF_MS_CONFIG),
+                streamsConfig.getInt(StreamsConfig.SEND_BUFFER_CONFIG),
+                streamsConfig.getInt(StreamsConfig.RECEIVE_BUFFER_CONFIG),
+                streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG), time);
+
+    }
+
+
+    /**
+     * Cretes a new topic with the given number of partitions and replication factor.
+     *
+     * @param internalTopicConfig
+     * @param numPartitions
+     * @param replicationFactor
+     */
+    public void createTopic(final InternalTopicConfig internalTopicConfig, final int numPartitions, final int replicationFactor, final long windowChangeLogAdditionalRetention) {
+        final Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
+        final Map<String, String> topicConfig = new HashMap<>();
+        for (String key : topicProperties.stringPropertyNames()) {
+            topicConfig.put(key, topicProperties.getProperty(key));
+        }
+        final CreateTopicsRequest.TopicDetails topicDetails = new CreateTopicsRequest.TopicDetails(numPartitions, (short) replicationFactor, topicConfig);
+        final Map<String, CreateTopicsRequest.TopicDetails> topics = new HashMap<>();
+        topics.put(internalTopicConfig.name(), topicDetails);
+
+        final CreateTopicsRequest createTopicsRequest = new CreateTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        final ClientResponse clientResponse = sendRequest(createTopicsRequest.toStruct(), ApiKeys.CREATE_TOPICS);
+        final CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(internalTopicConfig.name()).code() > 0) {
+            throw new StreamsException("Could not create topic: " + internalTopicConfig.name() + ". " + createTopicsResponse.errors().get(internalTopicConfig.name()).name());
+        }
+    }
+
+
+    /**
+     * Send a request to kafka broker of this client. Polls the request for a given number of iterations to receive the response.
+     *
+     * @param request
+     * @param apiKeys
+     */
+    private ClientResponse sendRequest(final Struct request, final ApiKeys apiKeys) {
+
+        final Node brokerNode = kafkaClient.leastLoadedNode(new SystemTime().milliseconds());
```

Comment:
Related to the comment below: since only one request is sent at a time, the `leastLoadedNode` function is not taking any load into consideration actually; so I think we could just iterate over the nodes and find one that is ready, and if all destination nodes are not ready, backoff based on the configured value and retry again.